gdal/frmts/hdf5/bagdataset.cpp

```diff
@@ -470,8 +471,15 @@ GDALDataset *BAGDataset::Open( GDALOpenInfo *poOpenInfo )
         return nullptr;
     }
 
+    // Buffer
+    void * pBuffer = nullptr;
+    off_t nSize = 0;
+    fprintf(stderr, "%s:%d\n", __FILE__, __LINE__);
+    hookup(poOpenInfo->pszFilename, &pBuffer, &nSize);
```

Comment:
shoud only be called when available (and probably when needed: for a regular file, perhaps using H5Fopen is a better choice), otherwise default to H5Fopen
Error status of hookup() should be checked